src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuilding.java

```diff
@@ -311,6 +317,26 @@ public void readFromNBT(@NotNull final NBTTagCompound compound)
 
         rotation = compound.getInteger(TAG_ROTATION);
         style = compound.getString(TAG_STYLE);
+
+        final String buildingName = compound.getString(TAG_BUILDING_TYPE);
+        final String md5 = compound.getString(TAG_SCHEMATIC_MD5);
+        final String structureName = Structures.SCHEMATICS_HUTS + '/' + style + '/' + this.getSchematicName() + buildingLevel;
+
+        if (Structures.getMD5(structureName).isEmpty())
```

Comment:
do you mean something like:
boolean hasMD5Hash(final String stuctureName) ?

fair point